Tests/Common/Communication/EventBusHandlerTests.swift

```diff
@@ -0,0 +1,565 @@
+        XCTAssertEqual(mockEventStorage.storeCallsCount, 0, "Event should not be stored if there are observers")
+    }
+
+    func test_postEvent_givenNoObservers_expectEventStoredInFile() async throws {
+        let eventBusHandler = initializeEventBusHandler()
+
+        // Given: A mock event and EventBus without observers
+        let event = TrackMetricEvent(deliveryID: String.random, event: String.random, deviceToken: String.random)
+
+        let postEventExpectation = XCTestExpectation(description: "Waiting for postEvent to complete")
+        mockEventBus.postClosure = { _ in
+            postEventExpectation.fulfill()
+            // Assume there are no observers
+            return false
+        }
+
+        // When: The event is posted
+        eventBusHandler.postEvent(event)
+
+        // Wait for the post operation to complete
+        await fulfillment(of: [postEventExpectation], timeout: 5.0)
+
+        // Then: Verify that post was called on EventBus and event was stored
+        XCTAssertEqual(mockEventBus.postCallsCount, 1, "post should be called once on EventBus")
+        XCTAssertEqual(mockEventStorage.storeCallsCount, 1, "Event should be stored if there are no observers")
+    }
+
+    func test_postEvent_givenTimestampedEvent_expectObserverReceivesCorrectTimestamp() async throws {
+        let eventBusHandler = initializeEventBusHandler()
+        let eventPostedTimestamp = Date()
+        let event = ResetEvent(timestamp: eventPostedTimestamp)
+
+        let eventReceivedExpectation = XCTestExpectation(description: "Event received by observer")
+        var receivedEventTimestamp: Date?
+
+        mockEventBus.postClosure = { event in
+            receivedEventTimestamp = event.timestamp
+            eventReceivedExpectation.fulfill()
+            return true
+        }
+
+        eventBusHandler.postEvent(event)
+
+        await XCTWaiter().fulfillment(of: [eventReceivedExpectation], timeout: 5.0)
+
+        XCTAssertEqual(receivedEventTimestamp, eventPostedTimestamp, "The timestamp received by the observer does not match the event's posted timestamp.")
+    }
+
+    func test_postEvent_givenMultipleObservers_expectSingleNotificationPerObserver() async throws {
+        let eventBusHandler = initializeEventBusHandler()
+        let event = RegisterDeviceTokenEvent(token: String.random)
+
+        // Simulate event caching for replay
+        mockEventCache.getEventReturnValue = [event]
+
+        // Expectations
+        let firstObserverReceivedExpectation = XCTestExpectation(description: "First observer should receive event only once")
+        firstObserverReceivedExpectation.expectedFulfillmentCount = 1
+        let secondObserverReceivedExpectation = XCTestExpectation(description: "Second observer should receive event only once")
+        secondObserverReceivedExpectation.expectedFulfillmentCount = 1
+
+        mockEventBus.postClosure = {
+            _ in
+            firstObserverReceivedExpectation.fulfill()
```

Comment:
why is this fulfilled here?

Reply:
since the real posting of events is done by the Notification center, the best we can do is make sure is `event bus` is calling `post` method.